### PR TITLE
[IMP] mail: remove messagingCreatedPromise from env

### PR DESCRIPTION
**Pull request by seb-odoo**
This is a bootstrapping promise that is otherwise completely useless in the env.

### CODE REVIEW

**Comment by seb-odoo** [comment]
@alexkuhn you can check this

File: addons/mail/static/src/messaging/messaging_env.js [hunk 1]
```diff
@@ -57,18 +51,16 @@ function addMessagingToEnv(env) {
...
-            await new Promise(resolve => setTimeout(resolve));
-            _addMessagingEntities(env);
         });
+        /**
+         * All JS resources are loaded, but not necessarily processed.
+         * We assume no messaging-related modules return any Promise,
+         * therefore they should be processed *at most* asynchronously at
+         * "Promise time".
+         */
+        await new Promise(resolve => setTimeout(resolve));
```
Comment (alexkuhn) [inline addons/mail/static/src/messaging/messaging_env.js:61]: `await new Promise(setTimeout);`

File: addons/mail/static/src/messaging/messaging_env.js [hunk 2]
```diff
@@ -43,12 +41,8 @@ function addMessagingToEnv(env) {
...
      * Messaging entities.
      */
-    let messagingCreatedPromise;
-    if (env.generateEntitiesImmediately) {
-        _addMessagingEntities(env);
-        messagingCreatedPromise = Promise.resolve();
-    } else {
-        messagingCreatedPromise = new Promise(resolve => {
+    if (!env.generateEntitiesImmediately) {
+        await new Promise(resolve => {
```
Comment (alexkuhn) [inline addons/mail/static/src/messaging/messaging_env.js:45]: is this promise really necessary?

Comment (seb-odoo) [reply addons/mail/static/src/messaging/messaging_env.js:45]: We have an event handler that is async by definition and we want to wait on it so yes? What do you suggest?

Comment (alexkuhn) [reply addons/mail/static/src/messaging/messaging_env.js:45]: oops misread, you can ignore this comment

File: addons/mail/static/src/messaging/messaging_env.js (hunk 1 above)
Comment (seb-odoo) [reply addons/mail/static/src/messaging/messaging_env.js:61]: This looks very tricky, I feel like this is asking for bugs to not have the promise constructor explicitly define its `resolve` but rely on the API of function define elsewhere.

**Review by alexkuhn — APPROVED** [review]

**Pull request merged**